### PR TITLE
Cache `bdbag` and `compact` manifest using object_key (#1476)

### PR DESCRIPTION
**Pull request by achave11**

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
This has a lot of the same commits as PR #1442. The two PR should probably be chained (branches rebased on top of each other and PR base branch adjusted).

File: test/service/test_manifest.py [hunk 1]
```diff
@@ -978,6 +981,7 @@ def test_full_metadata_missing_fields(self):
     @mock_sts
     @mock_s3
     @mock.patch('azul.service.manifest_service.ManifestService._get_seconds_until_expire')
+    @skip('Checking expiration date is no longer relevant')  # TODO: https://github.com/DataBiosphere/azul/issues/1507
```
Comment (jessebrennan) [inline test/service/test_manifest.py]: I had a look at this issue and didn't really understand the title.

File: test/service/test_manifest.py [hunk 2]
```diff
@@ -1089,38 +1093,50 @@ def test_manifest_content_disposition_header(self, _can_use_cached_manifest):
...
-        self.assertNotEqual(old_bundle_object_key, new_bundle_object_key)
-
-        # Updates or additions, unrelated to that project do not affect object key generation
-        self._index_canned_bundle(("f79257a7-dfc6-46d6-ae00-ba4b25313c10", "2018-09-14T133314.453337Z"))
-        generator = manifest_service.FullManifestGenerator(service, filters)
-        latest_bundle_object_key = service._derive_manifest_key(format_, filters, generator.manifest_content_hash)
-
-        self.assertEqual(latest_bundle_object_key, new_bundle_object_key)
+        for format_ in ['compact', 'full', 'terra.bdbag']:
+            with self.subTest(format_=format_):
```
Comment (jessebrennan) [inline test/service/test_manifest.py]:
```suggestion
            with self.subTest(msg='indexing new bundle', format_=format_):
```

File: test/service/test_manifest.py [hunk 3]
```diff
@@ -1089,38 +1093,50 @@ def test_manifest_content_disposition_header(self, _can_use_cached_manifest):
...
+                # and should remain valid ...
+                new_generator = manifest_service.ManifestGenerator.for_format(format_, service, filters)
+                self.assertEqual(old_bundle_object_key,
+                                 service._derive_manifest_key(format_, filters,
+                                                              new_generator.manifest_content_hash))
+                old_object_keys[format_] = old_bundle_object_key
+
+        new_object_keys = {}
+        for format_ in ['compact', 'full', 'terra.bdbag']:
+            with self.subTest(format_=format_):
```
Comment (jessebrennan) [inline test/service/test_manifest.py]:
```suggestion
            with self.subTest(msg='indexing second bundle', format_=format_):
```

File: test/service/test_manifest.py [hunk 4]
```diff
@@ -1089,38 +1093,50 @@ def test_manifest_content_disposition_header(self, _can_use_cached_manifest):
...
+                self._index_canned_bundle(new_bundle)
+                generator = manifest_service.ManifestGenerator.for_format(format_, service, filters)
+                new_bundle_object_key = service._derive_manifest_key(format_, filters,
+                                                                     generator.manifest_content_hash)
+                # ... invalidating the cached object previously used for the same filter.
+                self.assertNotEqual(old_object_keys[format_], new_bundle_object_key)
+                new_object_keys[format_] = new_bundle_object_key
+
+        for format_ in ['compact', 'full', 'terra.bdbag']:
+            with self.subTest(format_=format_):
```
Comment (jessebrennan) [inline test/service/test_manifest.py]:
```suggestion
            with self.subTest(msg='indexing unrelated bundle', format_=format_):
```

**Comment by hannes-ucsc** [comment]
At the moment this is still building, so I shouldn't be assigned.

**Pull request closed without merging**